s3_benchmark.py

```diff
@@ -0,0 +1,179 @@
+from awscrt.auth import AwsCredentialsProvider
+from awscrt.http import HttpHeaders, HttpRequest
+import time
+# import os
+import sys
+import threading
+
+GBPS = 1000 * 1000 * 1000
+
+
+class CrtLazyReadStream(object):
+    def __init__(self, filename, pattern, statistics, length=0):
+        self._filename = filename
+        self.length = length
+        self._stream = None
+        self._pattern = pattern
+        self._statistic = statistics
+
+    def _available_stream(self):
+        if self._stream is None:
+            self._stream = open(self._filename, self._pattern)
+            return
+
+        if self._stream.closed:
+            self._stream = open(self._filename, self._pattern)
+
+    def read(self, length):
+        self._available_stream()
+        data = self._stream.read(length)
+        read_len = len(data)
+        self._statistic.record_read(read_len)
+        if read_len is 0:
+            self._stream.close()
+        return data
+
+    def readinto1(self, m):
+        # Read into memoryview m.
+        self._available_stream()
+        len = self._stream.readinto1(m)
+        self._statistic.record_read(len)
+        if len is 0:
+            self._stream.close()
+        return len
+
+    def seek(self, offset, whence):
+        self._available_stream()
+        return self._stream.seek(offset, whence)
+
+    def close(self):
+        pass
+
+
+class Statistics(object):
+
+    def __init__(self):
+        self._lock = threading.Lock()
+        self.end_time = 0
+        self._bytes_peak = 0
+        self._bytes_avg = 0
+        self._bytes_read = 0
+        self._bytes_sampled = 0
+        self.sec_first_byte = 0
+        self.star_time = time.time()
+        self.last_sample_time = time.time()
+
+    def record_read(self, size):
+        with self._lock:
+            self._bytes_read += size
+            if self.sec_first_byte == 0:
+                self.sec_first_byte = time.time() - self.star_time
+            time_now = time.time()
+            if time_now - self.last_sample_time > 1:
+                bytes_this_second = (self._bytes_read - self._bytes_sampled) / (time_now - self.last_sample_time)
+                self._bytes_sampled = self._bytes_read
+                self._bytes_avg = (self._bytes_avg + bytes_this_second) * 0.5
+                if self._bytes_peak < bytes_this_second:
+                    self._bytes_peak = bytes_this_second
+                self.last_sample_time = time_now
+
+    def bytes_peak(self):
+        return (self._bytes_peak * 8) / GBPS
+
+    def bytes_avg(self):
+        return (self._bytes_avg * 8) / GBPS
+
+
+# Configurations
+region = "us-west-2"
+bucket_name = "aws-crt-python-s3-testing-bucket"
+object_name = "/0_10GB.txt"
+file_name = "." + object_name
+object_real_name = "/0_10GB"
+suffix = ".txt"
+repeat_times = 1
+bunch_size = 160
+
+writing_disk = True
+request_type = "download"
+
+# Initialization
+event_loop_group = EventLoopGroup(18)
```

Comment:
can we bind out the new numa aware event-loop-groups here and give this a go on node 0?